### ISSUE TITLE
Get some random bytes instead

### ISSUE THREAD

**Issue by andreaferretti** [nitram509/macaroons.js#13]
I don't know anything about how macaroons work, but I just read [this](https://github.com/nitram509/macaroons.js/blob/master/src/main/ts/CryptoTools.ts#L61-L63) on Hacker News.

```js
    /* XXX get some random bytes instead */
    var enc_plaintext:Buffer = new Buffer(MacaroonsConstants.MACAROON_HASH_BYTES);
    enc_plaintext.fill(0);
```

Maybe it is safe, but as depicted does not inspire confidence.

Also, a few lines later it seems to use a nonce filled with zeros, which is contrary to the concept of a nonce.

**Comment by nitram509** [comment]
Hi Andrea, thank you for pointing this out.
May I ask for the link to Hacker News?
I'm curious about the discussion.

**Comment by andreaferretti** [comment]
The discussion is [here](https://news.ycombinator.com/item?id=22354534)

In particular, I saw this library mentioned in [this comment](https://news.ycombinator.com/item?id=22356389)

**Comment by nitram509** [comment]
will be fixed in 0.3.9 release.
